lualib/resty/kong/tls.lua

```diff
@@ -121,6 +135,21 @@ function _M.disable_session_reuse()
 end
 
 
+local ssl_type = ffi.typeof("SSL*")
+local void_pp = ffi.new("void *[1]")
+
+function _M.get_ssl_pointer(sock)
+    local u = sock[SOCKET_CTX_INDEX]
+
+    local ret = kong_lua_kong_ffi_get_socket_ssl(u, void_pp)
+    if ret ~= NGX_OK then
+        return nil, "no ssl object"
+    end
+
+    return ffi.cast(ssl_type, void_pp[0])
```

Comment:
Should we localize `ffi.cast` for performance?

Reply:
modified